store/store.go

```diff
@@ -231,12 +243,16 @@ func (s *Store[H]) GetByHeight(ctx context.Context, height uint64) (H, error) {
 		return h, nil
 	}
 
+	return s.getByHeight(ctx, height)
+}
+
+func (s *Store[H]) getByHeight(ctx context.Context, height uint64) (H, error) {
```

Comment:
Have introduced a new method to get header by height. We don't need overcomplication as in `GetByHeight` 'cause we just want to read from the datastore.